pkg/config/marshal.go

```diff
@@ -0,0 +1,23 @@
+package config
+
+import "gopkg.in/yaml.v2"
+
+func (config Server) MarshalYAML() ([]byte, error) {
+	const sensitive = "<sensitive>"
+	if config.Auth.JWTSecret != "" {
+		config.Auth.JWTSecret = sensitive
+	}
+	if config.Auth.Github.ClientSecret != "" {
+		config.Auth.Github.ClientSecret = sensitive
+	}
+	if config.Auth.Google.ClientSecret != "" {
+		config.Auth.Google.ClientSecret = sensitive
+	}
+	if config.Auth.Gitlab.ClientSecret != "" {
+		config.Auth.Gitlab.ClientSecret = sensitive
+	}
+	if config.Auth.Internal.AdminUser.Password != "" {
+		config.Auth.Internal.AdminUser.Password = sensitive
+	}
```

Comment:
The fact that we have to manually add these here is scary to me. I'm worried that next time we add a sensitive parameter we'll forget to add it here. How can we avoid that? Maybe with annotations? That's that thing I was taking about, adding `yaml:"-"` to annotations. Would that work?